src/win/tty.c

```diff
@@ -1871,133 +1911,160 @@ static int uv_tty_write_bufs(uv_tty_t* handle,
               continue;
             }
 
-          } else if (utf8_codepoint == '?' && !(ansi_parser_state & ANSI_IN_ARG) &&
-                     handle->tty.wr.ansi_csi_argc == 0) {
-            /* Ignores '?' if it is the first character after CSI[. This is an
-             * extension character from the VT100 codeset that is supported and
-             * used by most ANSI terminals today. */
+          } else if (utf8_codepoint == '?' && prev_utf8_codepoint == '[') {
+            /* Ignores '?' if it is the first character after CSI */
+            /* This is an extension character from the VT100 codeset */
+            /* that is supported and used by most ANSI terminals today. */
+            ansi_parser_state |= ANSI_EXTENSION;
             continue;
 
-          } else if (utf8_codepoint >= '@' && utf8_codepoint <= '~' &&
-                     (handle->tty.wr.ansi_csi_argc > 0 || utf8_codepoint != '[')) {
-            int x, y, d;
+          /* SP SP is invalid */
+          } else if (utf8_codepoint == ' ' && prev_utf8_codepoint != ' ') {
+            /* Ignores SP. This has the possibility of set cursor style */
```

Comment:
I find this comment very cryptic. Why are you ignoring something which "has has some possibility"? I though set cursor style was now supported?